src/sinks/util/retries.rs

```diff
@@ -142,13 +145,16 @@ where
     }
 }
 
+// Safety: `L` is never pinned and we use no unsafe pin projections
```

Comment:
:+1: this is safe because there is no unsafe code needed to get around pin requirements